helios-services/src/main/java/com/spotify/helios/servicescommon/ZooKeeperAclProvider.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2016 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.servicescommon;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+import com.spotify.helios.servicescommon.coordination.Paths;
+
+import org.apache.curator.framework.api.ACLProvider;
+import org.apache.zookeeper.ZooDefs;
+import org.apache.zookeeper.data.ACL;
+import org.apache.zookeeper.data.Id;
+
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import static org.apache.zookeeper.ZooDefs.Perms.CREATE;
+import static org.apache.zookeeper.ZooDefs.Perms.READ;
+import static org.apache.zookeeper.ZooDefs.Perms.WRITE;
+import static org.apache.zookeeper.ZooDefs.Perms.DELETE;
+
+/**
+ * A ZooKeeper ACLProvider for Helios, allowing only read access for unauthenticated users. For
+ * "authenticated" access there's two different roles: agents and masters. Masters are granted
+ * all permissions (except ADMIN) to all nodes. Agents are granted the permissions it needs, to
+ * limit the consequences of the agent credentials being compromised.
+ * <p>
+ * Currently all agents share the same permissions. I.e. an agent can modify data that "belongs" to
+ * another agent (to the same extent that it can modify data that belongs to it).
+ * <p>
+ * Internally, the provider is implemented by granting permissions for users on paths based on
+ * regular expressions. The permissions for all rules that match a path apply. For example if there
+ * are two rules:
+ * <ol>
+ *   <li>"/foo", DELETE, user1</li>
+ *   <li>"/foo/bar", CREATE, user1</li>
+ * </ol>
+ * Then user1 can create and delete nodes on the /foo/bar node, but he can not delete nodes under
+ * the /foo node.
+ * <p>
+ * The provider uses ZooKeeper's "digest" ACL scheme.
+ */
+public class ZooKeeperAclProvider implements ACLProvider {
+
+  private static final String WILDCARD = "[^/]+";
```

Comment:
Yeah. A comment mentioning that it is meant to match a single path segment would probably suffice.